### DISCUSSION TITLE
Nextjs 13: What are some patterns for sharing state between client and server

### DISCUSSION THREAD

**Discussion by Mogu4419868** [vercel/next.js#49017] [Help]
### Summary

Project Structure:

```
app/
    [rsid]
        [categorySlug]
            [subCategorySlug]
                page.tsx
            layout.tsx
            template.tsx
            error.tsx
            loading.tsx
```
What I want to do:

I am fetching data dynamically inside page.tsx (server-side) using params.rsid. The data is a list of items and page.tsx defaults to display the first item in the list.

I fetch the same data inside layout.tsx and pass it to a client component heading.tsx. The heading.tsx component displays a dropdown menu of all the items in the list. When clicked, it should update page.tsx to show the selected item.

In other words, I want to display John, when John is selected from the dropdown menu and so on.

Are there any patterns that I should follow to achieve this? Than you in advance!

### Additional information

A visual example the code can be found here [Github codespace]( https://mogu4419868-automatic-xylophone-96r5vw7jqjq29vj4.github.dev) or here [Github Repo](https://github.com/Mogu4419868/temp)

**Comment by devjiwonchoi** [comment]
You can wrap the `heading.tsx` with the `layout.tsx` as a parent component in order to pass the `data` as `props`.

For example:

```js
// heading.tsx

'use client';

export default function Heading({title}: {title: string}){
  return (
    <div>
      <h1>{title}</h1>
    </div>
  );
}
```

```js
// layout.tsx

import Header from '@/components/Header';

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <html>
      <body>
        <Header title='some title' />
        {children}
      </body>
    </html>
  );
}
```

```js
// page.tsx

export default function Home() {
  return (
    <div>
      <h1>This is Home</h1>
    </div>
  );
}
```
Output:
![Screenshot 2023-06-03 at 11 04 28 PM](https://github.com/vercel/next.js/assets/120007119/0059126b-219a-4ba0-a1b0-1e1a56bde035)

**Reply by JanRuettinger** [reply]
What you recommend to store the state of the dropdown menu in layout or is there a better place?
